bin/pacignore

```diff
@@ -0,0 +1,238 @@
+  local pacignore_ignorepkg
+
+  # Proceed conditionally given previous exit code
+  if ! pacignore_ignorepkg="$(pacman-conf --config="$PACMAN_CONF" "IgnorePkg" 2>/dev/null)"; then
+    exit_code=1
+    {
+      gettext "Error in reading pacman configuration file"
+      printf "\n"
+    } >&2
+  else
+    # Convert string to array
+    mapfile -t "PACIGNORE_IGNOREPKG" < <(printf "%s" "$pacignore_ignorepkg")
+
+    # Proceed by pacignore subcommands
+    if [ "$PACIGNORE_SUBCOMMAND" == "ls" ]; then
+      printf "%s\n" "${PACIGNORE_IGNOREPKG[@]}"
+    else
+      for pkg; do
+        if ! "${PACIGNORE_SUBCOMMAND}_ignorepkg" "$pkg"; then
+          exit_code=1
+        else
+          success=1
+        fi
+      done
+
+      # Write to pacman configuration file if there is success
+      if ((success)) && [ "$PACIGNORE_SUBCOMMAND" != "check" ]; then
+        write_ignorepkg
+      fi
+    fi
+  fi
+
+  # Return consolidated exit code
+  return "$exit_code"
+}
+
+parse_options() {
+  # Parse the subcommand correctly
+  if [[ "$1" =~ ^(ls|check|add|rm)$ ]]; then
+    PACIGNORE_SUBCOMMAND="$1"
+    shift
+  fi
+
+  # Loop across all CLI arguments and categorize
+  while [[ -n "$1" ]]; do
+    case "$1" in
+      --pacman-conf)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN_CONF="$1"
+        else
+          {
+            gettext "Missing --pacman-conf argument"
+            printf "\n"
+            usage
+          } >&2
+          return 1
+        fi
+        ;;
+      -*)
+        local current_option
+        # shellcheck disable=SC2034
+        current_option="$1"
+        {
+          eval_gettext "Unrecognized option \$current_option"
+          printf "\n"
+          usage
+        } >&2
+        return 1
+        ;;
+      *)
+        pkgs=("$@")
+        break
+        ;;
+    esac
+    shift
+  done
+
+  # Check if any packages provided
+  if ((!"${#pkgs[@]}")) && [ "$PACIGNORE_SUBCOMMAND" != "ls" ]; then
+    {
+      gettext "No packages provided"
+      printf "\n"
+      usage
+    } >&2
+    return 1
+  fi
+}
+
+cli() {
+  # Parse all CLI options
+  parse_options "$@"
+
+  # Proceed with main workflow
+  main "${pkgs[@]}"
+}
+
+# Set script defaults
+PACMAN_CONF="/etc/pacman.conf"
+PACIGNORE_SUBCOMMAND="ls"
+
+# CLI execution (not for testing)
+if ((!LIB)); then
+  # Set bash -e flag for script termination on error
+  set -e
+
+  # Define pre-set locale
+  locale="$(dirname "$0")"/../share/locale
+
+  # Check for existence of locale
+  if [[ -d "$locale" ]]; then
+    # Packaged installation
+    TEXTDOMAINDIR=$(cd "$locale" && pwd)
+  else
+    # Probably testing ./pacignore
+    TEXTDOMAINDIR=/usr/share/locale
+  fi
+
+  # Export detected environmental variables
+  export TEXTDOMAIN=pacignore TEXTDOMAINDIR
+
+  # Check for help option
+  for arg; do
+    if [[ "$arg" =~ ^-h$|^--help$ ]]; then
+      usage
+      exit 0
+    fi
+  done
+
+  # Check if run as root
+  if ((UID)); then
```

Comment:
I don't know that we should explicitly forbid non-root for this. I would let file permissions errors raise themselves if they occur. An `ls` operation, for example, should work fine as non-root.

Reply:
Makes sense. Fixed this now